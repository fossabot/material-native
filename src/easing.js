'use strict';
import {Easing} from 'react-native';

export const standard = Easing.bezier(0.4, 0.0, 0.2, 1);
export const deceleration = Easing.bezier(0.0, 0.0, 0.2, 1);
export const acceleration = Easing.bezier(0.4, 0.0, 1, 1);
export const sharp = Easing.bezier(0.4, 0.0, 0.6, 1);
