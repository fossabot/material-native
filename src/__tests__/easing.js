'use strict';
import {mapValues} from 'lodash';
import * as easing from '../easing';

test('easing function results match snapshot', () => {
	const times = function *() {
		for ( let t = 0; t <= 10; ++t ) {
			yield t * 0.1;
		}
	};

	const results = mapValues(easing, (fn) => (
		Array.from(function *() {
			for ( const t of times() ) {
				yield fn(t);
			}
		}())
	));

	expect(results).toMatchSnapshot();
});
