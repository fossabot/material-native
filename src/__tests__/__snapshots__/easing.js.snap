// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`easing function results match snapshot 1`] = `
Object {
  "acceleration": Array [
    0,
    0.01837288420035122,
    0.06615137405647009,
    0.13593975189329927,
    0.2231667865925149,
    0.32481461060063765,
    0.4388214826130738,
    0.5637853298309402,
    0.6988430743785297,
    0.84375,
    1,
  ],
  "deceleration": Array [
    0,
    0.30384757729336814,
    0.5,
    0.6449902102692293,
    0.7552628175377846,
    0.839245057708503,
    0.9021108962296815,
    0.9473695577731094,
    0.9775593328553704,
    0.99460141820936,
    1,
  ],
  "sharp": Array [
    0,
    0.02133739266026176,
    0.08649724176651277,
    0.19415746864887784,
    0.3369323878604268,
    0.5,
    0.6630676121395734,
    0.8058425313511224,
    0.9135027582334873,
    0.9786626073397382,
    1,
  ],
  "standard": Array [
    0,
    0.025863014891412992,
    0.1339467750761098,
    0.3672957082859909,
    0.6135947465092803,
    0.7755613111161977,
    0.8759679094528104,
    0.9382454871360151,
    0.975267517565809,
    0.9943539287368762,
    1,
  ],
}
`;
